sdv/data_processing/data_processor.py

```diff
@@ -502,15 +502,11 @@ def _fit_hyper_transformer(self, data):
         Args:
             data (pandas.DataFrame):
                 Data to transform.
-            columns_created_by_constraints (set):
-                Names of columns that are not in the metadata but that should also
-                be transformed. In most cases, these are the fields that were added
-                by previous transformations which the data underwent.
 
         Returns:
             rdt.HyperTransformer
         """
-        if not self._hyper_transformer._fitted:
+        if not self._hyper_transformer._fitted or self._hyper_transformer._modified_config:
```

Comment:
If we just drop these conditions altogether, do any integration tests break? If not, maybe we should just do that